### PR TITLE
Make some ByteSizeValue instances always use the singleton

### PR DESCRIPTION
**Pull request by original-brownbear**
This comes out of a user heap dump investigation. In some snapshot corner cases we ran into about 100M of duplicate 0b instances.

-> even though it's a little heavy handed, lets make it so the common constants that we already have are used whenever possible.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java [hunk 1]
```diff
@@ -39,7 +39,16 @@ static class DeprecationLoggerHolder {
     public static final ByteSizeValue MINUS_ONE = new ByteSizeValue(-1, ByteSizeUnit.BYTES);
 
     public static ByteSizeValue ofBytes(long size) {
-        return new ByteSizeValue(size);
+        if (size == 0) {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java:42]: Could use switch here?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java:42]: can't switch on `long` :)

**Review by thecoop — APPROVED** [review]
Core change looks good

**Comment by original-brownbear** [comment]
Thanks Simon!

File: server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java:42]: doh

**Pull request merged**